### ISSUE TITLE
Pruning blocks

### ISSUE THREAD

**Issue by pigmej** [spacemeshos/pm#141]
More info: https://community.spacemesh.io/t/deferred-block-fetching-a-k-a-dangling-pointers/309

- [x] https://github.com/spacemeshos/go-spacemesh/issues/3607
- [x] https://github.com/spacemeshos/go-spacemesh/issues/3608
- [x] https://github.com/spacemeshos/go-spacemesh/issues/3609

**Comment by pigmej** [comment]
Currently blocked by clarifications in https://github.com/spacemeshos/pm/issues/144

**Comment by dshulyak** [comment]
issues 2 and 3 are important, but not blocking for genesis (which is reflected in the priority)
